WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -571,4 +571,96 @@
+      let!(:organizer2) { FactoryGirl.create(:user) }
+      let!(:competition1) { FactoryGirl.create(:competition, :registration_open, starts: 1.week.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+      let!(:competition2) { FactoryGirl.create(:competition, :registration_open, starts: 1.month.ago, organizers: [organizer2], eventSpecs: "222 333") }
+      let!(:competition3) { FactoryGirl.create(:competition, :registration_open, starts: 2.weeks.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+      let!(:competition4) { FactoryGirl.create(:competition, :registration_open, starts: 3.weeks.from_now, organizers: [organizer2], eventSpecs: "222 333") }
+      let!(:competition4) { FactoryGirl.create(:competition, :registration_open, starts: 2.month.ago, organizers: [organizer2], eventSpecs: "222 333") }
+      let!(:waldner_user) { FactoryGirl.create :user, name: "Jan-Ove Waldner" }
+      let!(:registration1) { FactoryGirl.create(:registration, competitionId: competition1.id, user: waldner_user) }
+      let!(:registration2) { FactoryGirl.create(:registration, competitionId: competition3.id, user: waldner_user) }
+      let!(:registration3) { FactoryGirl.create(:registration, competitionId: competition4.id, user: organizer1) }
+
+      before :each do
+        sign_in organizer1
+      end
+
+      it 'does show my upcoming competitions' do
+        get :my_competitions
+        expect(assigns(:not_past_competitions)).to eq [competition3, competition1]
+      end
+
+      it 'does show my past competitions' do
+        get :my_competitions
+        expect(assigns(:past_competitions)).to eq [competition4]
+      end
+    end
+
+    context 'when signed in as a delegate' do
+      let(:delegate1) { FactoryGirl.create(:delegate) }
+      let!(:organizer1) { FactoryGirl.create(:user) }
+      let!(:competition1) { FactoryGirl.create(:competition, :registration_open, starts: 1.week.from_now, organizers: [organizer1], delegates: [delegate1], eventSpecs: "222 333") }
+      let!(:competition2) { FactoryGirl.create(:competition, :registration_open, starts: 1.month.ago, organizers: [organizer1], delegates: [delegate1], eventSpecs: "222 333") }
+      let!(:competition3) { FactoryGirl.create(:competition, :registration_open, starts: 2.weeks.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+      let!(:competition4) { FactoryGirl.create(:competition, :registration_open, starts: 3.weeks.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+      let!(:competition4) { FactoryGirl.create(:competition, :registration_open, starts: 2.month.ago, organizers: [organizer1], eventSpecs: "222 333") }
+      let!(:waldner_user) { FactoryGirl.create :user, name: "Jan-Ove Waldner" }
+      let!(:registration1) { FactoryGirl.create(:registration, competitionId: competition1.id, user: waldner_user) }
+      let!(:registration2) { FactoryGirl.create(:registration, competitionId: competition3.id, user: delegate1) }
```

Comment:
Move all these `let!`s from all your tests level up, directly under the describe line. There's no reason to repeat them.